File: Ejercicio2/2.py
class Persona:
    def __init__(self, nombre, edad, salario, correo):
        self.nombre = nombre
        self.edad = edad
        self.salario = salario
        self.correo = correo

    def detalles(self):
        print("Nombre:", self.nombre)
        print("Edad:", self.edad)
        print("Correo electronico:", self.correo + "@EsteCorreoNoExiste.com")


class Empleado(Persona):
    def __init__(self, nombre, edad, salario, correo):
        super().__init__(nombre, edad, salario, correo)

    def detalles(self):
        super().detalles()
        print("Salario S/", self.salario)


class Cliente(Persona):
    def __init__(self, nombre, edad, correo):
        super().__init__(nombre, edad, 0, correo)

    def detalles(self):
        super().detalles()


print("---------Empleado---------")
empleado01 = Empleado("Gordon Freeman", 27, 12500, "gordonfre0m4n86")
empleado01.detalles()

print("---------Cliente---------")
cliente01 = Cliente("Juan Vasquez", 38, "juancitovazz8")
cliente01.detalles()

File: Ejercicio2/2.cpp
#include <iostream>
#include <string>
using namespace std;

class Persona
{
    public:
        string nombre;
        int edad;
        float salario;
        Persona(string _nombre, int _edad, float _salario, string _correo) : nombre(_nombre), edad(_edad),salario(_salario), correo(_correo){} //Constructor

    protected:
        string correo;

    void Detalles()
    {
        cout << "Nombre: " << nombre << endl;
        cout << "Edad " << edad << endl;
        cout << "Correo electronico: " << correo <<"@EsteCorreoNoExiste.com"<< endl;
    }
};

class Empleado : protected Persona
{
    public:
        Empleado(string _nombre, int _edad, float _salario, string _correo) : Persona(_nombre, _edad, _salario, _correo){} //Constructor
    void Detalles()
    {
        Persona::Detalles();
        cout << "Salario S/" << salario << endl;
    }
};

class Cliente : protected Persona
{
public:
    Cliente(string _nombre, int _edad, string _correo): Persona(_nombre, _edad, 0, _correo) {}

    void Detalles()
    {
        Persona::Detalles();
    }
};

int main()
{
    cout << "---------Empleado--------- " << endl;
    Empleado empleado01("Gordon Freeman",27,12500,"gordonfre0m4n86");
    empleado01.Detalles();

    cout << "---------Cliente--------- " << endl;
    Cliente cliente01("Juan Vasquez", 38, "juancitovazz8"); 
    cliente01.Detalles();

}